design/custom-ca-support.md

```diff
@@ -0,0 +1,107 @@
+# Custom CA Bundle Support for S3 Object Storage
+
+It is desired that Velero performs SSL verification on the Object Storage
+endpoint (BackupStorageLocation), but it is not guaranteed that the Velero
+container has the endpoints' CA bundle in it's system store. Velero needs to
+support the ability for a user to specify custom CA bundles at installation
+time and Velero needs to support a mechanism in the BackupStorageLocation
+Custom Resource to allow a user to specify a custom CA bundle. This mechanism
+needs to also allow Restic to access and use this custom CA bundle.
+
+## Goals
+
+- Enable Velero to be configured with a custom CA bundle at installation
+- Enable Velero support for custom CA bundles with S3 API BackupStorageLocations
```

Comment:
just to confirm here - you're proposing implementing support for this in two different ways (one at install time, for if you don't need different CA bundles per BSL, and one where you can have a different CA bundle per BSL), and the user could pick whichever approach works best for them, correct?

Reply:
Correct. Mainly because it would allow a user to specify clusterwide certs that they don't want to have to specify every time they create additional BSLs after Velero was installed.

I'm okay to reduce the complexity if you think managing it on the BSL alone is adequate.

Reply:
I think supporting both is :+1:, just wanted to confirm I was reading it correctly!